src/main/java/uk/ac/sanger/sccp/stan/service/register/filereader/BaseRegisterFileReader.java

```diff
@@ -0,0 +1,242 @@
+                problems.add("Unknown life stage: " + repr(lifeStageString));
+            }
+        }
+        return null;
+    }
+
+    public Address valueToAddress(Collection<String> problems, String addressString) {
+        if (!nullOrEmpty(addressString)) {
+            try {
+                return Address.valueOf(addressString);
+            } catch (IllegalArgumentException e) {
+                problems.add(e.getMessage());
+            }
+        }
+        return null;
+    }
+
+
+    /**
+     * Gets a cell value, trying to make it appropriate for the indicated type.
+     * Some conversions are possible between cell types and data types.
+     * For instance, a cell containing a number may be converted to a string;
+     * a cell containing a numeric (floating-point) value may be interpreted as an integer.
+     * @param type a type ({@link String} or {@link Integer})
+     * @param cell the cell to read
+     * @return the value read
+     * @param <T> the type of value read
+     * @exception IllegalArgumentException if the specified type cannot be read from the given cell
+     * @exception IllegalStateException may be thrown from the POI cells if they are of the wrong type
+     */
+    @SuppressWarnings("unchecked")
+    public <T> T cellValue(Class<T> type, Cell cell) {
+        CellType cellType = (cell==null ? null : cell.getCellType());
+        if (cellType==null || cellType==CellType.BLANK || type==Void.class) {
+            return null;
+        }
+        if (type== LocalDate.class) {
+            Date date = cell.getDateCellValue();
+            if (date==null) {
+                return null;
+            }
+            return (T) LocalDate.ofInstant(date.toInstant(), ZoneId.systemDefault());
+        }
+        if (type==Integer.class) {
+            if (cellType==CellType.NUMERIC || cellType==CellType.FORMULA) {
+                double value = cell.getNumericCellValue();
+                if (value%1!=0.0) {
+                    throw new IllegalArgumentException("Expected integer but got "+value);
+                }
+                return (T) (Integer) (int) value;
+            }
+            String value = cell.getStringCellValue();
+            try {
+                return (T) Integer.valueOf(value);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Expected integer but got "+repr(value));
+            }
+        }
+        if (cellType==CellType.NUMERIC) {
+            double value = cell.getNumericCellValue();
+            if (value%1==0.0) {
+                return (T) String.valueOf((int) value);
+            }
+            throw new IllegalArgumentException("Expected string but got number.");
+        }
+        String string = cell.getStringCellValue();
+        if (string!=null) {
+            string = string.trim();
+            if (string.isEmpty()) {
+                return null;
+            }
+        }
```

Comment:
This block can be replaced by the apache.commons.lang3 StringUtils.isEmpty method

Reply:
I could use StringUtils.isBlank() to avoid the trimming, but this method will return the trimmed string, so I want the 'trim()' call to be inside this method instead of somewhere else.